<?xml version="1.0" encoding="UTF-8"?>

<launch>
    <arg name = "vo_topic"             default = "/frontend/vo"/>
    <arg name = "lidar_odometry_topic" default = "/frontend/lidar_odometry/pose_relative"/>
    <arg name = "gnss_topic"           default = "/tf"/>
    <arg name = "imu_topic"            default = "/xsens/imu"/>

    <param name = "dataset"  type = "string" value = "milliAmpere" />
    <param name = "filename" type = "string" value = "seq03" />

    <node 
        name   = "backend"
        pkg    = "backend"
        type   = "backend_node"
        args   = "$(arg vo_topic) 
                  $(arg lidar_odometry_topic) 
                  $(arg gnss_topic) 
                  $(arg imu_topic)"
        output = "screen"
    >
    
    <remap from = "vo_topic"             to = "$(arg vo_topic)"/>
    <remap from = "lidar_odometry_topic" to = "$(arg lidar_odometry_topic)"/>
    <remap from = "gnss_topic"           to = "$(arg gnss_topic)"/>
    <remap from = "imu_topic"            to = "$(arg imu_topic)"/>

    </node>

</launch>